### PR TITLE
Request only data that has been updated since a certain date

### PR DESCRIPTION
**Pull request by pezholio**
Needed for https://github.com/theodi/shared/issues/487

### CODE REVIEW

File: spec/app_spec.rb [hunk 1]
```diff
@@ -200,4 +200,27 @@
...
+
+    Timecop.return
+
+    date = DateTime.now
+
+    5.times do
+      FactoryGirl.create(:address)
+    end
+
+    get 'addresses', { updated_since: date }
```
Comment (Floppy) [inline spec/app_spec.rb]: It's using a date object here, and in the params for the where. Should this be an ISO8601 string?

Comment (pezholio) [inline spec/app_spec.rb]: Good catch. It does auto transform it into an ISO8601 string, but I've added a `to_s` just to be more descriptive

File: spec/app_spec.rb [hunk 2]
```diff
@@ -219,4 +219,27 @@
...
+
+    Timecop.return
+
+    date = DateTime.now
+
+    5.times do
+      FactoryGirl.create(:address)
+    end
+
+    get 'addresses', { updated_since: date.to_s }
```
Comment (Floppy) [inline spec/app_spec.rb]: Still not sure this is precise enough; don't we want to support ISO8601 only, presumably? It seems vague, but perhaps it doesn't matter.

File: lib/ernest.rb [hunk 3]
```diff
@@ -62,7 +62,13 @@ def valid_key?
   get '/addresses' do
     content_type :json
 
-    page = Address.page(params[:page].to_i)
+    if params[:updated_since]
+      address = Address.where('updated_at >= ?', params[:updated_since])
```
Comment (Floppy) [inline lib/ernest.rb]: also slightly concerned here that there is no date parsing of the parameter to check validity. I could do `?updated_since=edballs` and cause application errors, presumably.

**Comment by pezholio** [comment]
OK, all done now :+1:

File: lib/ernest.rb [hunk 4]
```diff
@@ -62,19 +62,26 @@ def valid_key?
   get '/addresses' do
     content_type :json
 
-    page = Address.page(params[:page].to_i)
+    if params[:updated_since]
+      return 400 if !params[:updated_since].match(/[0-9]{4}\-[0-9]{2}\-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}\+[0-9]{4}/)
```
Comment (Floppy) [inline lib/ernest.rb]: sorry, this is probably really annoying, but this won't support partial ISO8601, or Z time zones. We can probably be a bit less strict here and just do a DateTime.parse; if that's not nil, then we're OK to go.

**Pull request merged**